vsphere/datadog_checks/vsphere/api.py

```diff
@@ -267,13 +276,21 @@ def get_infrastructure(self):
         root_folder = self._conn.content.rootFolder
         infrastructure_data[root_folder] = {"name": root_folder.name, "parent": None}
 
-        if self.config.should_collect_attributes:
+        if self.config.should_collect_attributes or self.config.collect_property_metrics:
             # Clean up attributes in infrastructure_data,
             # at this point they are custom pyvmomi objects and the attribute keys are not resolved.
 
             attribute_keys = {x.key: x.name for x in self._fetch_all_attributes()}
             for props in itervalues(infrastructure_data):
                 mor_attributes = []
+                all_properties = {}
+                if self.config.collect_property_metrics:
```

Comment:
nit - `all_properties` could be defined inside the `if` statement instead of outside to avoid creating it when not needed

Reply:
updated, good point!